### PR TITLE
Add missing property name in errors for draft 3

### PR DESCRIPTION
**Pull request by krishnaprasadmg**
Same as #91 the property names are missing in errors for draft 3 schema

### CODE REVIEW

**Comment by erayd** [comment]
Well spotted :-).

Please change the base branch for this PR to 6.0.0-dev, and use the new error handler.

File: src/JsonSchema/Constraints/UndefinedConstraint.php [hunk 1]
```diff
@@ -169,7 +169,7 @@ protected function validateCommonProperties(&$value, $schema = null, JsonPointer
             } elseif (isset($schema->required) && !is_array($schema->required)) {
                 // Draft 3 - Required attribute - e.g. "foo": {"type": "string", "required": true}
                 if ($schema->required && $value instanceof self) {
-                    $this->addError($path, 'Is missing and it is required', 'required');
+                    $this->addError($path, "The property " . $schema->name . " is required", 'required');
```
Comment (erayd) [inline src/JsonSchema/Constraints/UndefinedConstraint.php]: `$schema->name` does not exist, and causes test failures.
Noting this is the draft-03 boolean require you have added this for, the property name will not be available via `$schema` - you will need to obtain it from elsewhere.

File: src/JsonSchema/Constraints/UndefinedConstraint.php [hunk 2]
```diff
@@ -169,7 +169,8 @@ protected function validateCommonProperties(&$value, $schema = null, JsonPointer
             } elseif (isset($schema->required) && !is_array($schema->required)) {
                 // Draft 3 - Required attribute - e.g. "foo": {"type": "string", "required": true}
                 if ($schema->required && $value instanceof self) {
-                    $this->addError($path, 'Is missing and it is required', 'required');
+                    $propertyName = current($path->getPropertyPaths());
+                    $this->addError($path, "The property " . $propertyName . " is required", 'required');
```
Comment (erayd) [inline src/JsonSchema/Constraints/UndefinedConstraint.php]: Per my previous comments, please alter this line to use the new error handler. The syntax you are currently using is deprecated and from 6.0.0 is no longer supported.

**Comment by krishnaprasadmg** [comment]
Sorry for the confusion, I've created a new PR, I'll close this. Thank you for looking into this!

**Pull request closed without merging**